src/MainWindow.ui

```diff
@@ -868,6 +871,18 @@
     <string>F6</string>
    </property>
   </action>
+  <action name="designAction3DPrint">
+   <property name="icon">
+    <iconset resource="../openscad.qrc">
+     <normaloff>:/images/3dprint-32.png</normaloff>:/images/3dprint-32.png</iconset>
+   </property>
+   <property name="text">
+    <string>&amp;3D Print</string>
+   </property>
+   <property name="shortcut">
+    <string>F7</string>
```

Comment:
Let's not make a shortcut for this. It's not something you'll every do very often or casually, and it's too easy to mis-click when indenting to hit F6.

Reply:
For some people (with complicated geometry) render is actually faster then preview. Sometimes you need to render the design to see how certain preview artifacts are resolved.

Accidental uploads could be a privacy/confidentiality concern to some people.
You could open a can of worms regarding copyright.

Calling it "3D Print" is also a bit misleading, when it send the data to the internet (instead of to the 3D printer software/slicer).

One side note: The inputdriver automatically creates the action list. We could expand our input drivers to make it possible to map free/reserverd keyboard buttons to actions.

Reply:
I missed the shortcut when looking though the pull request earlier, here's my thinking for completeness :-).....

I'd be happy with F7 provided the following possible future development
- We add a send to print service preferences pane which has some means to disable the feature.
- We add a local OctoPrint, realistically there's at least a similar number of people with their own printer who may want to send to their own OctoPrint and not to an external service.

>  If they do end up accidentally press the button that could be a good way for them to discover that this new functionality even exists and plant a little seed to maybe try it later. It shouldn't take more than 5 seconds of their time to close a browser window.

That gave me a pause actually. I think that is not a good way of discovering this type of features. Considering also that this is uploading user data to an external service, we may need to have some "first start" info (not just talking GDPR compliance but also considering IP issues).

Reply:
> Usually I don't render unless I'm about to print something. I use the preview for everything up until that point because it's an order of magnitude faster than rendering. 

I think that is unusual. Often you will want to render/export and examine the STL before you print, a few times.

> I don't think I've "accidentally" pressed a function key even once in the last year. The function row is way at the top 

You have one of them old style keyboards, not very hip...

> options re-implementing the natural F5 F6 F7 combo

Actually for me preview, render, export would be natural, maybe F8 for Print?

I have my print machine isolated, so like someone* else mentioned, Fn-key to export to a default location with my USB stick is good. * in nopheads filename/directory stuff.

This just shows how everyone has their UI/UX quirks. 
So until Michael's idea of mapping keys, perhaps leave out the shortcut ATM.  

-----

I would absolutely NOT want a browser to open if I accidentally hit a key. (REALLY HATE those S/W updates that think opening a browser so they can show me something - and meanwhile track me - is.....) anyway
I agree with tp "preferences pane which has some means to disable the feature." except "...means to **enable** the feature", ie I don't think the default setup should point to network externalities, that is something the user should enable with intent. By all means make it easy to setup, for example a pre-populated drop-down or such.

> We add a local OctoPrint

I have yet to play with Octoprint, but if it is vanilla enough, or has its config in standard locations this would be good to add to that drop-downn list.

Reply:
The export happens automatically in this code, so I think F7 is the natural "3D Print" shortcut, if we decide to have one.

Reply:
Adding entire dialog and such for enabling or disabling one specific feature that should just blend in with everything else doesn't really seem value-add to me.  We'd be adding complexity for something that MIGHT happen and will probably very rarely happen, and where the downside of it happening isn't really that big - a couple wasted seconds.  This isn't a "Send $1000 from my bank account to ISIS" hotkey, it's just a streamlined way to print your 3d object.  ;)  LOL  Regarding IP - Print a Thing is bound by their terms of service to keep all files confidential, and their suppliers are bound to confidentiality as well.

My preferences, in order from most to least:

1.  Leave the hotkey in there, as is, and use the tool like that ourselves during a testing phase.  See how often we accidentally hit F7, and if it ACTUALLY ends up being a problem, let's solve the problem then.
2.  Do this initial release without the F7 hotkey.  :(  Reserve F7 for this feature (nothing else should use this- is there a way to enforce this?) in the future.  Come back to this in 3 months when the whole thing seems a lot less scary.

Reply:
Ctrl+p is traditionally to print the window content to a paper printer, while OpenSCAD doesn't currently print the editors content (or the console or preview pane for that matter) I don't think this standard function should be messed with (IMO). Unless a dialogue box is to be used, which could later have options to paper-print these things??

Perhaps Shift-Ctrl-p?

Reply:
> makes perfect sense

https://xkcd.com/1172/

For your use case. Not mine. 
No way do I do Preview -> Render -> 'send something unexamined to a printer - particularly if it going to cost me real money'.

Personally I would like F7 to be export to STL or better for the future Export Dialogue that has been discussed from time to time.

MHO.

Reply:
Sending it to Print a Thing doesn't cost any money.  It just uploads it to a site where you can preview the part in 3D and then add to to your cart.  There are about 5 different steps where you can choose to stop what you're doing.  Pressing F7 spends nothing.

Reply:
You are viewing this solely from your perspective, that the only thing anyone would ever want to do after rendering is to upload it to your print service. Fare enough that it what you want to promote.

I'm saying there are other use cases which also make perfect sense. While I hope you all success, I believe that upload will always be a niche case. As such using such a prominent key is not merited when there are more holistic core functions which suit. IMHO. But I am not the king of UI.

Reply:
I can see the case for F7=export.  But I would think that if that was going to happen it would have happened by now?  I think the reason might be because there are lots of different types of export, so it's hard to apply one shortcut to them.  stl, off, amf, etc.  I'd be ok with F7=some "select which filetype you'd like to export" dialog, and F8=3DPrint, if it comes to that.  I'm guessing such a dialog gets rid of the convenience that a "shortcut" normally offers, though.

But I still think this just looks polished and makes sense:

https://ibb.co/S61Pcd7

Ultimately, this might be one thing where two intelligent and well meaning people might still disagree.  Nothing against either of us.

Reply:
An Export dialogue box has been on the books for some time #649.